controllers/hvpa_controller.go

```diff
@@ -413,8 +457,21 @@ func (r *HvpaReconciler) reconcileHpa(hvpa *autoscalingv1alpha1.Hvpa) (*autoscal
 		return nil, err
 	}
 
-	err = r.Create(context.TODO(), hpa)
-	return hpa.Status.DeepCopy(), err
+	if !r.IsAutoscalingV2Enabled {
+		err = r.Create(context.TODO(), hpa)
+		return hpa.Status.DeepCopy(), err
+	}
+	v2Hpa, err := r.Convert_v2beta1_HPA_to_v2(hpa)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.Create(context.TODO(), v2Hpa)
+	v2beta1Hpa, err := r.Convert_v2_HPA_to_v2beta1(v2Hpa)
+	if err != nil {
+		return nil, err
+	}
+	return v2beta1Hpa.Status.DeepCopy(), err
```

Comment:
This is related to a previous question in this review, and depends on its response. If the conversion function is guaranteed to return an independent copy which does not reference the input object, DeepCopy() would not be necessary here.